wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1004,13 +1016,10 @@ impl PhysicalDeviceProperties {
                 extensions.push(ext::descriptor_indexing::NAME);
             }
 
-            // Require `VK_KHR_shader_float16_int8` and `VK_KHR_16bit_storage` if the associated feature was requested
-            if requested_features.contains(wgt::Features::SHADER_F16) {
+            // Always require `VK_KHR_shader_float16_int8` if available as it enables Int8
+            // optimizations. It is also _needed_ if `wgt::Features::SHADER_F16` is requested.
+            if self.supports_extension(khr::shader_float16_int8::NAME) {
```

Comment:
I think this should be 
```rust
requested_features.contains(wgt::Features::SHADER_F16) ||
self.supports_extension(khr::shader_float16_int8::NAME)
```

to maintain current behavior where we get an error if `SHADER_F16` was requested but `VK_KHR_shader_float16_int8` is not available.

Reply:
Ah, makes sense.